docs/guidelines/gh-actions-runners.md

```diff
@@ -0,0 +1,65 @@
+# GitHub Actions and Self-Hosted Runners Guideline
+
+The scope of this guideline is to provide generic security advice for GitHub Actions and specific security advice for using self-hosted runners.
+
+In general, our first advice to teams would be to create a **Threat Model for the Software Development Lifecycle**.
+
+## GitHub Actions in General
+
+When using GitHub Actions, it is good practice to:
+
+- Investigate and familiarize yourself with the [Security Guides](https://docs.github.com/en/actions/security-guides) provided by GitHub and take appropriate actions.
+- Investigate and align on repository configuration (repository settings)
+    - Actions -> General
+        - Action permissions
+            - The default should be "Disable actions" if GitHub Actions are not used.
+            - The default should be "Allow enterprise, and select non-enterprise, actions and reusable workflows" if GitHub Actions are used.
+            - "Allow actions created by GitHub" can be checked.
+            - Add the specific actions you are going to use into the "Allow specified actions and reusable workflows" section
+        - "Fork pull request workflows from outside collaborators"
+            - The default should be "Require approval for first-time contributors".
+  - All external actions that you plan to use should be examined/audited. This could include examining the repos and using services like [StepSecurity's Action Advisor](https://app.stepsecurity.io/action-advisor).
+- Pin the versions of actions that you are using (use SHA).
+- Use OpenID Connect for access to Azure Resources.
```

Comment:
If this is the federated option, a suggestion could be to maybe provide a link or append what will be gained by using OpenID Connect (no need to add secrets like `client_id` / `client_secret`):
```
- Prefer OpenID Connect for access to Azure Resources over client_id/client_secret
```
or similar phrasing.. with link to e.g. https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation

Reply:
It is. I will links to indicate this.